pkg/runctl/controller.go

```diff
@@ -363,12 +362,10 @@ func (c *Controller) syncHandler(key string) error {
 		pipelineRun.UpdateResult(api.ResultErrorInfra)
 		pipelineRun.StoreErrorAsMessage(err, "failed to load configuration for pipeline runs")
 		c.metrics.CountResult(pipelineRun.GetStatus().Result)
-
+		runManager.Cleanup(pipelineRun)
```

Comment:
I didn't search all places where `finish` is called. If `Cleanup` is always called before `finish`, shall we incorporate them into one function?

Reply:
I want to do this in #248 to keep merge conflicts as small as possible.